### ISSUE TITLE
PropsChildren's type isn't exactly Array

### ISSUE THREAD

**Issue by nghuuphuoc** [japgolly/scalajs-react#282]
I have a component that can pass children nodes which can be empty:

```
object Hello {
  private val ComponentName = this.getClass.getSimpleName

  private case class Props(message: String)

  private case class Backend(scope: BackendScope[Props, _]) {
    def render(P: Props, C: PropsChildren): vdom.ReactTagOf[Div] = {
      println(C)
      // If C is empty then render an empty div
      if (C.isEmpty) {
        <.div()
      } else {
        <.div(^.cls := "message", 
          C,
          P.message
        )
      }
    }
  }

  private val component = ReactComponentB[Props](ComponentName)
    .stateless
    .renderBackend[Backend]
    .build

  def apply(message: String)(children: ReactNode*): ReactComponentU[_, _, _, TopNode] = {
    component(Props(message), children: _*)
  }
}
```

My attempt to use is

```
Hello("message here")() // Is this correct way to pass an empty child?
Hello("message here")(<.div("Children"))
```

It compiles successfully, but when loading the page on browser, I got the exception for the 1st usage (`Hello("message here")()`):

![issue](https://cloud.githubusercontent.com/assets/4226036/15111428/2d7476a6-1611-11e6-8f30-0b6e473e01c5.png)

I added `println(C)` inside the `Backend.render()` method, and I got `undefined` when calling (`Hello("message here")()`).

**Comment by nghuuphuoc** [comment]
Updated:
The current workaround is to use `React.Children.count(C)`.

**Comment by japgolly** [comment]
Sounds like the PropsChildren type should be `UndefOr[Array[?]]`. Problem is that the docs say it's an opaque type but then in JS usage people treat it as an array. Ok, will need a fix and a few tests.

**Comment by japgolly** [comment]
Oh and @nghuuphuoc to answer your question, yes, using `React.Children.count` to check if empty is a good idea.

**Comment by japgolly** [comment]
This will be fixed as part of #259.

**Comment by japgolly** [comment]
Fixed on `topic/neo` branch.
